zerver/management/commands/soft_deactivate_users.py

```diff
@@ -65,6 +72,26 @@ def handle(self, *args: Any, **options: str) -> None:
 
             users_activated = do_soft_activate_users(users_to_activate)
             logger.info('Soft Reactivated %d user(s)' % (len(users_activated)))
+        elif catch_up:
+            if user_emails:
+                users_to_catch_up = UserProfile.objects.filter(
+                    email__in=user_emails,
+                    **filter_kwargs
+                )
```

Comment:
Should both this code path and the `activate` code path be also filtering for `long_term_idle=True`?  Seems like a valuable perf optimization.

Reply:
Yep! Except, I guess the error handling needs to be a little more careful - and we need to separately print the users who didn't need to be activated or caught up and the email IDs which were invalid.

Reply:
Punting on this, since we don't expect the code path where emails are explicitly supplied to have more than a handful of emails.